File: app/src/main/java/igrek/forceawaken/domain/task/AwakeTask.java
package igrek.forceawaken.domain.task;

import igrek.forceawaken.activity.AwakenActivity;

/**
 * Task to do by an user after waking up
 */
public interface AwakeTask {
	
	/**
	 * @return instance cloned from a prototype
	 */
	AwakeTask getInstance();
	
	default double getProbabilityWeight() {
		return 1;
	}
	
	void run(AwakenActivity activity);
}

File: app/src/main/java/igrek/forceawaken/task/AwakeTask.kt
package igrek.forceawaken.task

import igrek.forceawaken.activity.AwakenActivity

/**
 * Task to do by an user after waking up
 */
interface AwakeTask {
    /**
     * @return instance cloned from a prototype
     */
    val instance: AwakeTask

    val probabilityWeight: Double
        get() = 1.0

    fun run(activity: AwakenActivity)
}